Make Endpoints deployable

jmaupetit opened the pull request:
## Purpose

When a `Service` depends on an `Endpoint` (for non-trashable environments), they must share the same name to be usable by `Pod`s. The counterpart is that some DeploymentConfigs related to this service should only be created in trashable environments. The current work improves the situation.

## Proposal

- [x] ignore empty rendered templates
- [x] make endpoints deployable (B/G)
- [x] update `richie` templates following the aforementioned conventions
- [x] update `edxapp` templates following the aforementioned conventions

jmaupetit committed:
⚡️(templates) filter app rendered empty templates

Our wait loop during deployment is based on the fact that for every
DeploymentConfig, we should have a running pod. If some DCs are not
created under certain conditions (e.g. env_type), our wait loop is
infinite and thus broken. To solve this issue, we filter templates that
are empty when rendered so that the object list we cook only contains
objects that will get created.

jmaupetit committed:
⚡️(endpoints) create endpoints during deployment

Some services are trashable given particular environments. This is the
case for some database services. As a service name should match the
related endpoint name, we choose to deploy endpoints instead of creating
them only during the project initialization (with B/G support).

Bonus: this will avoid potential collisions of previously shared
endpoints between apps from the same project.

jmaupetit committed:
⚡️(richie) improve postgresql service deployment

We now only create a DC for the postgreqsl service if required
(trashable environment), and make the service point to the related
endpoint.

jmaupetit committed:
🔧(main) update postgresql endpoint IP address

Previous IP address was a simple mock, let's use a real one.

jmaupetit committed:
🐛(richie) add blue-green prefix to ALLOWED_HOST

ALLOWED_HOST django setting was not taking into account the host
blue-green prefix. This is now fixed.

jmaupetit committed:
⚡️(edxapp) improve database services deployment

We now only create a DC for the mongo and mysql services if required
(trashable environment), and make the services point to the related
endpoint.

Sustancial changes have been made to the way we handle settings for the
lms and cms services as database hosts change between deployments.

sampaccoud commented on group_vars/customer/patient0/development/configs/edxapp/cms/settings.yml.j2:
```diff
@@ -2,14 +2,16 @@
 
 PLATFORM_NAME: "{{ project_display_name }}"
 
+SITE_NAME: "{{ project_display_name }}"
```
The SITE_NAME is used as the domain to build absolute urls included in emails...

You should remove all mentions to it in Arnold as it is now configured with a good default in fun-platform: 
https://github.com/openfun/fun-platform/blob/master/config/lms/docker_run_production.py#L193

sampaccoud commented on group_vars/customer/patient0/development/configs/edxapp/lms/settings.yml.j2:
```diff
@@ -2,14 +2,16 @@
 
 PLATFORM_NAME: "{{ project_display_name }}"
 
+SITE_NAME: "{{ project_display_name }}"
```
ditto

sampaccoud commented on tasks/get_objects_for_app.yml:
```diff
@@ -2,13 +2,31 @@
   tags: deploy
 
+# Remove empty rendered templates (e.g. inactivated for an env_type)
+- name: Ignore empty rendered templates
```
:crying_cat_face:

jmaupetit replied:
Yes, we should work on this...

jmaupetit replied:
I've declared an issue for this pain point, see #128

jmaupetit committed:
🔧(settings) remove SITE_NAME setting

SITE_NAME is now properly defined in fun-platform, hence, there is no
need to redefine it.

jmaupetit committed:
⚡️(acme) create openshift-acme role for the project

First we thought it was a good idea to create the openshift-acme role
globally (cluster level), but it meant losing control on it. Hence, we
roll back our changes to create this role and bind it to a service
account we control.

jmaupetit committed:
⬆️'edxapp) upgrade edxapp to hawthorn.1-1.0.6

This new release fixes a bug related to email settings.

jmaupetit committed:
🔧(edxapp) add support for blue-green host urls

The future fun-platform release will allow us to override the
ALLOWED_HOSTS settings via the config utils. Hence, we've added every
url each service (cms & lms) can be queried from.

jmaupetit committed:
🔧(edxapp) add email settings for patient0-development

This work adds example email settings for patient0-development using
mailcatcher.

sampaccoud commented on core_apps/acme/templates/app/role.yml.j2:
```diff
@@ -50,16 +50,3 @@ rules:
   - update
```
in commit message: s/loose/lose
